librz/core/cmd/cmd_search.c

```diff
@@ -226,7 +226,8 @@ RZ_IPI RzCmdStatus rz_cmd_info_gadget_handler(RzCore *core, int argc, const char
 	}
 
 	RzRopSearchContext *context = rz_core_rop_search_context_new(core, argv[1], false, RZ_ROP_GADGET_PRINT, state);
-	return rz_core_rop_gadget_info(core, context);
+	RzCmdStatus status = rz_core_rop_gadget_info(core, context);
```

Comment:
This is unnecessary

Reply:
Or did you want to add also the `_context_free()` function here?

Reply:
`_context_free` happens from rz_core_rop_gadget_info. Maybe i could directly return the function status here. :+1:

Reply:
You forgot to change this one